lib/providers/firestore_provider.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:qrdemo/providers/qr_provider.dart';
+import 'package:qrdemo/providers/routing_provider.dart';
+import 'package:qrdemo/providers/timer_provider.dart';
+import 'package:qrdemo/styles/textstyles.dart';
+
+class FirestoreProvider extends ChangeNotifier {
+  List<String>? _codes = [];
+  List<String>? get codes => _codes;
+
+  FirestoreProvider() {
+    FirebaseFirestore.instance;
+  }
+
+  void confirmScan(Barcode result, BuildContext context) {
```

Comment:
Another way of doing this could be to build the UI based on the return on the confirmScan. For example, if confirmScan returned a boolean, and true if Barcode result is not empty, and false if the Barcode result is empty. We can then do a conditional statement in the UI that confirmScan ? showSnackBar(...confirmed scan...) : showSnackBar(...code expired...)

Reply:
I went ahead and added the Snackbar widget in the scanner screen and have the logic return a future boolean so it can await those results. Once the qr reader listener listens for a scan, it'll take the string from the barcode and check if it is empty or not and open a snackbar depending on the string.